python/cuml/experimental/cluster/robust_single_linkage.pyx

```diff
@@ -0,0 +1,256 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# distutils: language = c++
+
+from libc.stdint cimport uintptr_t
+from libcpp cimport bool
+
+from cython.operator cimport dereference as deref
+
+import numpy as np
+import cupy as cp
+
+from cuml.common.array import CumlArray
+from cuml.common.base import Base
+from cuml.common.doc_utils import generate_docstring
+from cuml.raft.common.handle cimport handle_t
+from cuml.common import input_to_cuml_array
+from cuml.common.array_descriptor import CumlArrayDescriptor
+from cuml.common.mixins import ClusterMixin
+from cuml.common.mixins import CMajorInputTagMixin
+
+from cuml.metrics.distance_type cimport DistanceType
+
+from .hdbscan_plot import SingleLinkageTree
+
+cdef extern from "cuml/cluster/hdbscan.hpp" namespace "ML::HDBSCAN::Common":
+    cdef cppclass robust_single_linkage_output[int, float]:
+        robust_single_linkage_output(const handle_t &handle,
+                                     int n_leaves,
+                                     int *labels,
+                                     int *children,
+                                     int *sizes,
+                                     float *deltas,
+                                     int *mst_src,
+                                     int *mst_dst,
+                                     float *mst_weights)
+        int get_n_leaves()
+        int get_n_clusters()
+
+    cdef cppclass RobustSingleLinkageParams:
+        int k
+        int min_samples
+        int min_cluster_size
+
+        float alpha
+
+        float cluster_selection_epsilon,
+
+        bool allow_single_cluster,
+
+
+cdef extern from "cuml/cluster/hdbscan.hpp" namespace "ML":
+
+    void robust_single_linkage(const handle_t &handle,
+                               const float *X,
+                               size_t m,
+                               size_t n,
+                               DistanceType metric,
+                               RobustSingleLinkageParams &params,
+                               robust_single_linkage_output &out)
+
+
+def delete_output(obj):
```

Comment:
Not entirely sure it's super worth it, but both `delete_outputs` could be condensed to one